### PR TITLE
lambda fix for single unbracketed arguments

### LINKED ISSUE #73: Arguments to lambda functions wrapped in brackets even with only one argument
```python
sql = """
transform(array('a','b','c'), X -> upper(X))
"""

import sqlglot
sqlglot.transpile(sql)
```
```
> ["TRANSFORM(ARRAY('a', 'b', 'c'), (X) -> UPPER(X))"]
```

I discovered this as it was breaking some pyspark queries. Minimal pyspark example below:
```python
from pyspark.context import SparkContext
from pyspark.sql import SparkSession
sc = SparkContext.getOrCreate()
spark = SparkSession(sc)
from pyspark.sql import Row
data_list = [
    {"array_col": ['a', 'b']},

        ]

df = spark.createDataFrame(Row(**x) for x in data_list)
df.createOrReplaceTempView("df")

import sqlglot 
sql = "select transform(array_col, x -> x) as transformed from df"
spark.sql(sql).toPandas()

sql = sqlglot.transpile(sql, read='spark', write='spark')
spark.sql(sql).toPandas()
```

Will submit a PR to:
- distinguish between `x -> f(x)` and `(x,y,z) -> f(x,y,z)` syntax
- add a new test 
- update `identity.sql` accordingly

### PR DESCRIPTION
**Pull request by samnlindsay**
PR to fix #73 

I know this works for the specific use cases I'm aware of where the desired (or at least allowed) syntax is always `x ->`. 

If there is a dialect where `x ->` is incorrect and only `(x) ->` is allowed then this generic application of the fix may not work.

### CODE REVIEW

**Comment by tobymao** [comment]
is this required or just aesthetic? if its just for looks, i may not be inclined to merge it

File: tests/test_transpile.py [hunk 1]
```diff
@@ -280,3 +280,13 @@ def test_error_level(self, logger):
 
         with self.assertRaises(ParseError):
             transpile("x + 1 (")
+
+    def test_lambda(self):
```
Comment (tobymao) [inline tests/test_transpile.py:284]: can you delete these unit tests, they aren't necessary, or simply add these variants to identity.sql

File: sqlglot/generator.py [hunk 2]
```diff
@@ -423,8 +423,11 @@ def join_sql(self, expression):
         return f"{expression_sql}{op_sql} {this_sql}{on_sql}"
 
     def lambda_sql(self, expression):
+        a, b = (
```
Comment (tobymao) [inline sqlglot/generator.py:426]: can we clean this up a bit

args = self.expressions(expression, flat=True)
args = f"({args})" if len(args) > 1 else args
return self.no_identify(f"{args} -> ...")

File: tests/test_transpile.py (hunk 1 above)
Comment (samnlindsay) [reply tests/test_transpile.py]: Thanks, I thought that might be the case. I wrote the test before I discovered the extensive list in identity.sql

File: sqlglot/generator.py (hunk 2 above)
Comment (samnlindsay) [reply sqlglot/generator.py]: Much neater, thanks. I wasn't happy with my code but no alternative came to mind.

**Review by tobymao — APPROVED** [review]

**Pull request merged**